src/index.ts

```diff
@@ -74,13 +74,16 @@ export const sanitize = (kennitala: string): string | undefined =>
 export const formatKennitala = (
   kennitala: string,
   spacer: boolean = true
-): string | undefined => {
-  const kt = sanitizeInput(kennitala);
-  if (!kt) return undefined;
-  return `${kt.slice(0, 6)}${spacer ? "-" : ""}${kt.slice(6)}`;
```

Comment:
I think it makes sense to check the input with sanitizeInput before adding the spacer. Could just return kennitala unchanged if it does not look like a kennitala.

Reply:
That's essentially what the format function is doing now. Removes non-digits using regex and returns the output with the spacer in the right place if the string is long enough.

The idea was that people can use this to format input fields

onChange->format

So as you type it would add the spacer automatically, perhaps useful for display purposes in an input field.

What do you think?

Reply:
I usually prefer a special masking input to provide a better UX when editing parts of the value in the field.

It might be confusing for users to see `formatKennitala('accidentally a phone number 8693179')` return `869317-9`, so maybe it should just return empty string or the input string if there's not a full kennitala in there.

But I don't have strong opinions either way, as long as it is well documented and always has a string output for a string input. 👍

Reply:
Good point most people will probably just use input masking. I have the sanitize function available when you need to both validate and remove non digit characters for storing in a normalized db for example.

I added some clarification in the `README.md` on the purpose of the `.format` function.